### DISCUSSION TITLE
2.7.4

### DISCUSSION THREAD

**Discussion by matidau** [Z-Hub/Z-Push#98] [Announcements]
## What's Changed
* Update version for autodiscover subdirectory by @matidau in https://github.com/Z-Hub/Z-Push/pull/78
* Fixed Folders created on Outlook 2021 get duplicated by @matidau in https://github.com/Z-Hub/Z-Push/pull/80
* fix: error `PHP message: PHP Fatal error:  Uncaught Error: Attempt to assign property "body" on null` by @dallaslu in https://github.com/Z-Hub/Z-Push/pull/81
* Update mime_encode.php by @dallaslu in https://github.com/Z-Hub/Z-Push/pull/82
* fixed #88 PGP and S/MIME encryption by @dallaslu in https://github.com/Z-Hub/Z-Push/pull/89

## New Contributors
* @dallaslu made their first contribution in https://github.com/Z-Hub/Z-Push/pull/81

**Full Changelog**: https://github.com/Z-Hub/Z-Push/compare/2.7.3...2.7.4

<hr /><em>This discussion was created from the release <a href='https://github.com/Z-Hub/Z-Push/releases/tag/2.7.4'>2.7.4</a>.</em>

**Comment by delacroix1983** [comment]
When using with php 8.3 how to get php-mapi? Seems like it is not updated to php 8.3?

**Comment by matidau** [comment]
If you're running the Kopano Backend you are stuck with 7.4
https://github.com/Z-Hub/Z-Push/discussions/54

If you aren't running Kopano, change your config to not use Kopano
https://github.com/Z-Hub/Z-Push/discussions/42

**Reply by delacroix1983** [reply]
Thank you it is now working.
Any idea how to resolve the following or how to trace back? iPhone is not syncing because of error.
Ignored broken message (SyncMail). Reason: '2' Folderid: '028864bd' message id '36748'

**Reply by matidau** [reply]
I didn't think that prevent syncing, was more to let you know that you have a message that it can't send to the mobile.

Try adjusting your log level to see if you can get more info. If it is a large mailbox you will also need to give it time to perform the first sync.

**Comment by delacroix1983** [comment]
That is the only error it gives. Looks like a message from GoogleGroup causing the problem.

> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] -------- Start
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] cmd='Sync' devType='iPhone' devId='9hun9hrnnh4m7778vs1lvgu1c0' getUser='em@ma.il' from='89.134.14.129' version='2.7.4
> ' method='POST'
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] Request::ProcessHeaders() ASVersion: 14.0
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] ZPush::CommandNeedsProvisioning(0): true
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] ASDevice data loaded for user: 'em@ma.il'
> ....
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] LoopDetection->Detect(): case 3.3 detected - in loop mode, increase loop counter
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] LoopDetection->Detect(): case 3.3.1 detected - broken item should be next, attempt to ignore it - id '36748'
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] LoopDetection->Detect(): loop data: loopcount(4), maxCount(81), queued(231), ignored(36747)
> 23/10/2024 07:41:50 [128956] [WARN] [em@ma.il] Mobile loop detected! Messages sent to the mobile will be restricted to 2 items in order to identify the conflict
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] BackendIMAP->StatMessage('028864bd','36747')
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] BackendIMAP->getImapIdFromFolderId('028864bd') = INBOX
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] BackendIMAP->GetMessage('028864bd', '36747', '1,2,4')
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] BackendIMAP->getImapIdFromFolderId('028864bd') = INBOX
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] BackendIMAP->StatMessage('028864bd','36747')
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] BackendIMAP->getImapIdFromFolderId('028864bd') = INBOX
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] Utils::getRawMailHeaders(): subject = 
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] Utils::getRawMailHeaders(): from = =?UTF-8?Q?=27M=C3=A1ty=C3=A1s_Sch=27_via_=2D_SZVB?= <xxxx@googlegroups.com>
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] BackendIMAP->GetMessage(): Message is multipart: 1, smime: 0, smime encrypted: 0
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] BackendIMAP->GetMessage(): getBodyPreferenceBestMatch: 2
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] BackendIMAP->GetMessage():: iOS device iPhone->Apple-iPhone15C2/2201.3370 detected
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] BackendIMAP->GetMessage(): after thinking a bit we will use: 4
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] StringStreamWrapper::stream_open(): initialized stream length: 7 - HTML-safe-truncate: false
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] StringStreamWrapper::stream_open(): initialized stream length: 24609 - HTML-safe-truncate: false
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] LoopDetection->IgnoreNextMessage(): this should be the broken one, but is not! Resetting loop count.
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] LoopDetection->IgnoreNextMessage(): Loop mode, potential broken message id '36747'
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] LoopDetection->SetBrokenMessage('028864bd', '36747'): tracking broken message
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] BackendIMAP->StatMessage('028864bd','36748')
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] BackendIMAP->getImapIdFromFolderId('028864bd') = INBOX
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] BackendIMAP->GetMessage('028864bd', '36748', '1,2,4')
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] BackendIMAP->getImapIdFromFolderId('028864bd') = INBOX
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] BackendIMAP->StatMessage('028864bd','36748')
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] BackendIMAP->getImapIdFromFolderId('028864bd') = INBOX
> **23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] Utils::getRawMailHeaders(): subject = 
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] Utils::getRawMailHeaders(): from = =?UTF-8?Q?=27M=C3=A1ty=C3=A1s_Sch=27_via_=2D_SZVB?= <xxxxx@googlegroups.com>**
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] BackendIMAP->GetMessage(): Message is multipart: 1, smime: 0, smime encrypted: 0
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] BackendIMAP->GetMessage(): getBodyPreferenceBestMatch: 2
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] BackendIMAP->GetMessage():: iOS device iPhone->Apple-iPhone15C2/2201.3370 detected
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] BackendIMAP->GetMessage(): after thinking a bit we will use: 4
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] StringStreamWrapper::stream_open(): initialized stream length: 7 - HTML-safe-truncate: false
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] StringStreamWrapper::stream_open(): initialized stream length: 28863 - HTML-safe-truncate: false
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] LoopDetection->IgnoreNextMessage(): previously broken message '36748' is still broken and will not be tracked anymore
> **23/10/2024 07:41:50 [128956] [ERROR] [em@ma.il] Ignored broken message (SyncMail). Reason: '2' Folderid: '028864bd' message id '36748'**
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] ImportChangesStream->ImportMessageChange('36748'): message ignored and requested to be removed from mobile
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] HandleSync(): Exported maxItems of messages: 2 / 231
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] FileStateMachine->SetState() written 15758 bytes on file: '/var/lib/z-push/0/c/9hun9hrnnh4m7778vs1lvgu1c0-dd8443f8-5cbd-44ed-bb20-e347b2580262-57'
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] SyncCollections->SaveCollection(): Data of folder '028864bd' changed
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] FileStateMachine->SetState() written 1051 bytes on file: '/var/lib/z-push/0/c/9hun9hrnnh4m7778vs1lvgu1c0-dd8443f8-5cbd-44ed-bb20-e347b2580262-fd'
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] WBXMLEncoder->endTag() WBXML output completed
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] BackendIMAP->close_connection(): disconnected from IMAP server
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] FileStateMachine->SetState() written 777 bytes on file: '/var/lib/z-push/0/c/9hun9hrnnh4m7778vs1lvgu1c0-bs-1729624830'
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] DeviceManager->Save(): Device data changed
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] FileStateMachine->SetState() written 5901 bytes on file: '/var/lib/z-push/0/c/9hun9hrnnh4m7778vs1lvgu1c0-devicedata'
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] DeviceManager->Save(): Device data saved
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] LoopDetection->ProcessLoopDetectionTerminate()
> 23/10/2024 07:41:50 [128956] [ INFO] [em@ma.il] cmd='Sync' memory='2.63 MiB/4.00 MiB' time='0.48s' devType='iPhone' devId='9hun9hrnnh4m7778vs1lvgu1c0' getUser='em@ma.il' from='89.134.14.129' idle='0s' version='2.7.4
> ' method='POST' httpcode='200'
> 23/10/2024 07:41:50 [128956] [DEBUG] [em@ma.il] -------- End

**Comment by matidau** [comment]
Do you have the headers for that google groups message? Looks like we are having trouble parsing them.

And what Imap server/package are you using?

**Comment by delacroix1983** [comment]
X-MC-Unique: 8cjaJ6NpMDOcNuAg1pwW-A-16
From: =?UTF-8?Q?=27M=C3=A1ty=C3=A1s_Sch=27_via_=2D_SZVB?= <xxxxx@googlegroups.com>
To: "xxxxx@googlegroups.com" <xxxxx@googlegroups.com>
Subject: =?UTF-8?B?W1Bhc2thbFJvc2UgLSBTWlZCXSBIUlRMLW5layBrw7xsZMO2dHQgdMOpcnRpdmV2w6lueQ==?=
	=?UTF-8?B?ZXMgbGV2ZWxlaw==?=
Thread-Topic: =?Windows-1252?Q?HRTL-nek_k=FCld=F6tt_t=E9rti=E9?=

Postfix / Dovecot combination.

**Reply by matidau** [reply]
Headers look fine to me, might be the rest of the message at fault. Any chance you can test with a different mailbox that doesn't have the google groups messages? Just want to make sure there isn't a configuration issue not related to these emails.
